hack/configapi/cmd/root.go

```diff
@@ -85,21 +86,43 @@ func runCmd(cmd *cobra.Command, _ []string) error {
 		return fmt.Errorf("unmarshalling version file: %w", err)
 	}
 
-	latestAPIVersion, err := attestationconfigapi.NewFetcher().FetchAzureSEVSNPVersionLatest(ctx)
+	dateStr, err := cmd.Flags().GetString("upload-date")
 	if err != nil {
-		return fmt.Errorf("fetching latest version: %w", err)
+		return fmt.Errorf("getting upload date: %w", err)
 	}
-
-	isNewer, err := isInputNewerThanLatestAPI(inputVersion, latestAPIVersion.AzureSEVSNPVersion)
-	if err != nil {
-		return fmt.Errorf("comparing versions: %w", err)
+	var uploadDate time.Time
+	if dateStr != "" {
+		uploadDate, err = time.Parse("2006-01-01-01-01", dateStr)
+		if err != nil {
+			return fmt.Errorf("parsing date: %w", err)
+		}
+	} else {
+		uploadDate = time.Now()
 	}
-	if isNewer || force {
-		if force {
-			cmd.Println("Forcing upload of new version")
-		} else {
+
+	doUpload := false
+	if !force {
+		latestAPIVersion, err := attestationconfigapi.NewFetcher().FetchAzureSEVSNPVersionLatest(ctx, time.Now())
+		if err != nil {
+			return fmt.Errorf("fetching latest version: %w", err)
+		}
+
+		isNewer, err := isInputNewerThanLatestAPI(inputVersion, latestAPIVersion.AzureSEVSNPVersion)
+		if err != nil {
+			return fmt.Errorf("comparing versions: %w", err)
+		}
+		if isNewer {
 			cmd.Printf("Input version: %+v is newer than latest API version: %+v\n", inputVersion, latestAPIVersion)
+		} else {
+			cmd.Printf("Input version: %+v is not newer than latest API version: %+v\n", inputVersion, latestAPIVersion)
 		}
+		doUpload = isNewer
+	} else {
+		doUpload = true
+		cmd.Println("Forcing upload of new version")
+	}
+
+	if doUpload {
```

Comment:
```suggestion
		doUpload = isNewer
	} 

	if doUpload || force {
```

if you want the print statement you could add a `if force {cmd.Println...}`

trying to remove a few `else` blocks.